raiden/tests/utils/factories.py

```diff
@@ -1067,6 +1085,69 @@ def make_transfers_pair(
     )
 
 
+@dataclass
+class TestChainState:
+    chain_state: ChainState
+    our_address: Address
+    payment_network_address: PaymentNetworkAddress
+    token_address: TokenAddress
+    token_network_address: TokenNetworkAddress
+    channel_set: ChannelSet
+
+    @property
+    def channels(self):
+        return self.channel_set.channels
+
+
+def make_chain_state(
+    number_of_channels: int,
+    properties: List[NettingChannelStateProperties] = None,
+    defaults: NettingChannelStateProperties = NettingChannelStateProperties.DEFAULTS,
+) -> TestChainState:
+    channel_set = make_channel_set(
+        number_of_channels=number_of_channels, properties=properties, defaults=defaults
+    )
+    assert (
+        len(set(c.canonical_identifier.token_network_address for c in channel_set.channels)) == 1
+    )
+    assert len(set(c.our_state.address for c in channel_set.channels)) == 1
+    token_network_address = channel_set.channels[0].canonical_identifier.token_network_address
+    token_address = make_address()
+
+    token_network = TokenNetworkState(
+        address=token_network_address, token_address=token_address, network_graph=None
+    )
+    for netting_channel in channel_set.channels:
+        token_network.channelidentifiers_to_channels[
+            netting_channel.canonical_identifier.channel_identifier
+        ] = netting_channel
+        token_network.partneraddresses_to_channelidentifiers[
+            netting_channel.partner_state.address
+        ].append(netting_channel.canonical_identifier.channel_identifier)
+
+    payment_network_address = make_address()
+    our_address = channel_set.channels[0].our_state.address
+
+    chain_state = ChainState(
+        pseudo_random_generator=random.Random(),
+        block_number=1,
+        block_hash=make_block_hash(),
+        our_address=our_address,
+        chain_id=UNIT_CHAIN_ID,
+    )
+    chain_state.identifiers_to_paymentnetworks[payment_network_address] = PaymentNetworkState(
+        address=payment_network_address, token_network_list=[token_network]
+    )
+    return TestChainState(
```

Comment:
To be honest, this feels counterintuitive. This sounds more to me as a Test chain data container. However, i would be very much against doing this (like we did with the property classes) and go for properly populating a chain state with the required test data such as payment / token networks and channels.

Reply:
I added a docstring to `make_chain_state` and renamed the container to make the intentions clearer.